### ISSUE TITLE
Storing trusted hosts in registry breaks macOS and Linux support

### ISSUE THREAD

**Issue by ChrisMagnuson** [darkoperator/Posh-SSH#205]
https://github.com/darkoperator/Posh-SSH/blob/master/Trust.ps1 would need to be refactored so that it uses something like a directory and files to store these.

https://github.com/darkoperator/Posh-SSH/issues/167 would likely need to be merged first and if support for this needs to be included in the solution.

Is this something that your open to getting a pull request for or would a fork of the module to support no windows workloads be preferred?

**Comment by darkoperator** [comment]
Yes actually a person from MS has done a major refactor and Im waiting on his PR to have it all working in Linux and OS X

**Comment by callifo** [comment]
Hi, is there a workaround in the meantime? I've attempted to run a script across multiple hosts with pwsh and posh and been trying to work out what the error meant when run on my Linux hosts:

New-SSHSession : The type initializer for 'Microsoft.Win32.Registry' threw an exception.
At line:1 char:1
+ New-SSHSession -ComputerName 172.16.2.153
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [New-SSHSession], TypeInitializationException
+ FullyQualifiedErrorId : System.TypeInitializationException,SSH.NewSshSession

Couldn't find anyone else with the same issue but looks like this might be it?

**Comment by darkoperator** [comment]
No workaround since before starting it loads the key mapping from the registry in to memory. The module was designed and written before .net core and PowerShell core.

**Comment by darkoperator** [comment]
Well nit so much of a fix but a full refactor :)

**Comment by ChrisMagnuson** [comment]
@darkoperator Is there a link to where the refactor/rewrite is being done?

**Comment by ChrisMagnuson** [comment]
Bump, we are still expanding our use of PowerShell Core and need to leverage ssh and would love to use a module for that. @darkoperator Any word on the refactor or a location we can get a beta version of this code?

**Comment by darkoperator** [comment]
Not at this moment. Project depends on what spare time I have to work on it vs other projects I currently have.

**Comment by darkoperator** [comment]
Hope is to start working on it in the second half of October with the latest renci beta when it comes out.

**Comment by ChrisMagnuson** [comment]
@darkoperator I was under the impression from above that someone else had already done this and you were waiting on a pull request but it sounds like now your thinking your going to have to write this yourself which is unfortunate and certainly a larger undertaking.

Thanks for the update.

**Comment by darkoperator** [comment]
Sadly the person has not replied to mu emails or messages so I will have to do it my self.

**Comment by SX86** [comment]
Hi! Are we seeing any progress on this issue? I just ran into it. Can't open a SSH session inside a PowerScript that is run in a Docker container.

Thank you!

**Comment by darkoperator** [comment]
Trying to get a version of the lib to compile without problems in .net core.

**Comment by Toh3mi** [comment]
Hi, if interested, 3par-powershell-toolkit uses your posh-ssh to connect to 3par. I also managed to set up 3par monitoring by scripts, but customers have a serious interest in monitoring work on linux powershell (about 50+ installations).
I look forward to your update with impatience to advance in my work. I wish you success !!!

**Comment by nillocftw** [comment]
I'm having the same issue running mcr.microsoft.com powershell:centos-7 docker image when creating an SSH session using Posh-SSH@2.2 `New-SSHSession` 

`New-SSHSession: The type initializer for 'Microsoft.Win32.Registry' threw an exception.`

**Comment by darkoperator** [comment]
The module is a Windows PowerShell module only, it is not written for Mac OS or Linux

**Comment by brunoscota** [comment]
I`m facing the same problem because, as Toh3mi said, we're interested in 3par automation.

**Comment by brunoscota** [comment]
> Hi, if interested, 3par-powershell-toolkit uses your posh-ssh to connect to 3par. I also managed to set up 3par monitoring by scripts, but customers have a serious interest in monitoring work on linux powershell (about 50+ installations).
> I look forward to your update with impatience to advance in my work. I wish you success !!!

I'm using your scripts to monitor my 3par with zabbix, but i'm facing this problem.

**Comment by brunoscota** [comment]
3.0.0-beta2 working fine for me now!! forget it.. uhaeuahuahe
